src/tools/bin/plot_chargeConservation_overTime.py

```diff
@@ -111,22 +108,43 @@ def deviation_charge_conservation(h5file):
     for field_name in f["/data/{}/fields/".format(timestep)].keys():
         if field_name[0:8] == "Density_":
             # load species density
-            species_Density = np.array(f["/data/{}/fields/".format(timestep) + field_name])
+            species_Density_pointer = f["/data/{}/fields/".format(timestep) + field_name]
+            species_Density = np.array(species_Density_pointer)
             # choose norm to be the maximal charge density of all species
             norm = np.max([norm, np.amax(np.abs(species_Density))])
             # add charge density to total charge density
             charge += species_Density
+            if species_Density_pointer.attrs['_size'][2] < 2:
```

Comment:
It checks the attribute `_size` of any/all `density_[species]`. `libSplash` always keeps this as a array of length 3. It describes the size of the data in each dimension. If we are in a **2D** simulation, the size of the `z`/`[2]`-component is `1`/`<2`. The code changes the 2D3D flag if one Density data set is 2D.